src/plugins/formActions/pluginSpec.js

```diff
@@ -0,0 +1,212 @@
+ * licenses. See the Open Source Licenses file (LICENSES.md) included with
+ * this source code distribution or the Licensing information page available
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+import {
+    createMouseEvent,
+    createOpenMct,
+    resetApplicationState
+} from 'utils/testing';
+
+import { debounce } from 'lodash';
+
+describe('EditPropertiesAction plugin', () => {
+    let editPropertiesAction;
+    let openmct;
+    let element;
+
+    beforeEach((done) => {
+        element = document.createElement('div');
+        element.style.display = 'block';
+        element.style.width = '1920px';
+        element.style.height = '1080px';
+
+        openmct = createOpenMct();
+        openmct.on('start', done);
+        openmct.startHeadless(element);
+
+        editPropertiesAction = openmct.actions.getAction('properties');
+    });
+
+    afterEach(() => {
+        editPropertiesAction = null;
+
+        const activeOverlays = openmct.overlays.activeOverlays;
+        activeOverlays.forEach(overlay => overlay.dismiss());
+
+        return resetApplicationState(openmct);
+    });
+
+    it('editPropertiesAction exists', () => {
+        expect(editPropertiesAction.key).toEqual('properties');
+    });
+
+    it('edit properties action applies to only persistable objects', () => {
+        spyOn(openmct.objects, 'isPersistable').and.returnValue(true);
+
+        const domainObject = {
+            name: 'mock folder',
+            type: 'folder',
+            identifier: {
+                key: 'mock-folder',
+                namespace: ''
+            },
+            composition: []
+        };
+        const isAppliesTo = editPropertiesAction.appliesTo([domainObject]);
+        expect(isAppliesTo).toBe(true);
+    });
+
+    it('edit properties action does not applies to non persistable objects', () => {
+        spyOn(openmct.objects, 'isPersistable').and.returnValue(false);
+
+        const domainObject = {
+            name: 'mock folder',
+            type: 'folder',
+            identifier: {
+                key: 'mock-folder',
+                namespace: ''
+            },
+            composition: []
+        };
+        const isAppliesTo = editPropertiesAction.appliesTo([domainObject]);
+        expect(isAppliesTo).toBe(false);
+    });
+
+    it('edit properties action when invoked shows form', (done) => {
+        const domainObject = {
+            name: 'mock folder',
+            notes: 'mock notes',
+            type: 'folder',
+            identifier: {
+                key: 'mock-folder',
+                namespace: ''
+            },
+            modified: 1643065068597,
+            persisted: 1643065068600,
+            composition: []
+        };
+
+        editPropertiesAction.invoke([domainObject])
+            .catch(() => {
+                done();
+            });
+
+        setTimeout(() => {
```

Comment:
`setTimeout` is a code smell. Is there some other way that we can test this? What's causing the asynchrony here, is it Vue? If so, could we use `Vue.nextTick()` instead?

Reply:
yes, I tried to use `Vue.nextTick()` but was facing issue with first render of form was setting all values to default then setting to actual values and was missing to capture value change in `Vue.nextTick()`

Reply:
OK, this might be happening because [the EditPropertiesAction calls the formsAPI](https://github.com/nasa/openmct/blob/3c247334767ed84f37dae9eb2061c18ee3469799/src/plugins/formActions/EditPropertiesAction.js#L101), which [mounts a new Vue component to render the form](https://github.com/nasa/openmct/blob/3c247334767ed84f37dae9eb2061c18ee3469799/src/api/forms/FormsAPI.js#L115). That Vue component [then mounts another Vue component _from its `mounted` function_](https://github.com/nasa/openmct/blob/3c247334767ed84f37dae9eb2061c18ee3469799/src/api/forms/components/FormRow.vue#L107). So, the inner components don't even start to get mounted until the first tick has  happened.

So, waiting 2 ticks might fix it, but that's really hacky. Here are a couple of other alternatives that I can think of:

* `FormRow` [emits an `onChange` event on mount](https://github.com/nasa/openmct/blob/master/src/api/forms/components/FormRow.vue#L104). Maybe you could spy on the `onChange` listener passed in with the call to the form API? 
* `FormRow` [also calls `formControl.show`](https://github.com/nasa/openmct/blob/master/src/api/forms/components/FormRow.vue#L107) when it's mounted, which is part of the public API, so we could perhaps use Jasmine to spy on that to know when it's called?
* Finally we _could_ add an `onShow` handler to the form that is emitted when the form is fully rendered.